### PR TITLE
resource/alicloud_db_instance: Adds several attributes to enable ssl, like ssl_enabled, ca_type , server_cert, client_ca_cert  and so on

### PR DESCRIPTION
**Pull request by zhaoguolong-sy2**

### CODE REVIEW

File: alicloud/resource_alicloud_db_readonly_instance.go [hunk 1]
```diff
@@ -166,14 +205,112 @@ func resourceAlicloudDBReadonlyInstanceUpdate(d *schema.ResourceData, meta inter
...
+
+			if v, ok := d.GetOk("acl"); ok && v.(string) != "" {
+				request["ACL"] = v.(string)
+			}
+
+			if v, ok := d.GetOk("replication_acl"); ok && v.(string) != "" {
+				request["ReplicationACL"] = v.(string)
+			}
+		}
+		request["ConnectionString"] = instance["ConnectionString"]
```
Comment (xiaozhu36) [inline alicloud/resource_alicloud_db_readonly_instance.go:278]: Is there no need to distinguish the public or private connection string?

**Review by xiaozhu36 — CHANGES_REQUESTED** [review]

File: alicloud/resource_alicloud_db_readonly_instance.go [hunk 2]
```diff
@@ -110,10 +110,49 @@ func resourceAlicloudDBReadonlyInstance() *schema.Resource {
 				Optional: true,
 				Computed: true,
 			},
-			"force_restart": {
-				Type:     schema.TypeBool,
+			"ssl_action": {
```
[the remark this answers is not in the document]
Comment (zhaoguolong-sy2) [reply alicloud/resource_alicloud_db_readonly_instance.go:113]: ok, I'll retain this attribute.

File: alicloud/resource_alicloud_db_readonly_instance.go [hunk 3]
```diff
@@ -166,14 +205,112 @@ func resourceAlicloudDBReadonlyInstanceUpdate(d *schema.ResourceData, meta inter
...
+			request["SSLEnabled"] = 0
+		}
+		if sslAction == "Open" {
+			request["SSLEnabled"] = 1
+		}
+		if sslAction == "Update" {
+			request["SSLEnabled"] = 2
+		}
+
+		if sslAction == "Update" && d.Get("engine").(string) == "PostgreSQL" {
```
[the remark this answers is not in the document]
Comment (zhaoguolong-sy2) [reply alicloud/resource_alicloud_db_readonly_instance.go:231]: becase master instance use ssl_action, so stay the same.

File: alicloud/resource_alicloud_db_readonly_instance.go (hunk 1 above)
Comment (zhaoguolong-sy2) [reply alicloud/resource_alicloud_db_readonly_instance.go:278]: yes, it's no need.

**Pull request merged**